core/primitives/src/epoch_manager.rs

```diff
@@ -120,6 +123,24 @@ impl AllEpochConfig {
         )
     }
 
+    pub fn from_epoch_config_store(
+        chain_id: &str,
+        epoch_length: BlockHeightDelta,
+        epoch_config_store: EpochConfigStore,
+    ) -> Self {
+        let genesis_epoch_config = epoch_config_store.get_config(PROTOCOL_VERSION).as_ref().clone();
+        Self {
+            config_store: Some(epoch_config_store),
+            chain_id: chain_id.to_string(),
+            epoch_length,
+            // The fields below must be DEPRECATED.
```

Comment:
cool. can you add a TODO referring to the issue id, so we do not forget it? (same for other todos and deprecations)